src/test/java/com/jwplayer/southpaw/SouthpawEndToEndTest.java

```diff
@@ -126,37 +128,13 @@ public SouthpawEndToEndTest(
             }
         }
 
+        int entryChecks = 0;
```

Comment:
how about something like this? Everything below could be removed

```suggestion
        assertEquals(denormalizedRecords.entrySet().stream()
            .flatMap(entry -> entry.getValue().entrySet().stream())
            .peek(innerEntry ->
                assertEquals(
                    expectedResults.get(innerEntry.getKey()),
                    innerEntry.getValue()
                )).count(), expectedResults.size());
```